### DISCUSSION TITLE
HIPAA Compliance / Ability to Sign BAA

### DISCUSSION THREAD

**Discussion by croossin** [vercel/community#690] [Feature Question] [answered]
It would be awesome to deploy software that requires BAA's and/or HIPAA compliance.

**Comment by swarnava** [comment]
This is something already in our radar. Thank you for reporting.

**Comment by shivpratik** [comment]
It would be great to deploy healthcare applications on Vercel with HIPAA compliance.

**Reply by leerob** [reply]
Hey! Are there specific parts of HIPAA you're looking for compliance for? Could you share more about your needs? Being compliant might depend on the workload you're wanting to run on Vercel.

**Reply by murad-alqadi** [reply]
@leerob Any updates on this? We are also looking to deploy a healthcare application on Vercel.

**Reply by Erik-Koning** [reply]
@leerob To become HIPAA compliant, vercel has to meet a set of security practices. There is not really a definitive list anywhere but Mongodb has an excellent summary here about the key items they addressed to gain HIPAA compliance: https://www.mongodb.com/blog/post/making-hipaa-compliant-applications-mongodb

Based on this page: https://vercel.com/docs/concepts/deployments/deployment-protection#vercel-authentication
It seems Vercel is meeting standards for authentication, and we know 2FA is available, so thats great.

Based on this page: https://vercel.com/docs/concepts/security
It seems Vercel is meeting HIPAA for encryption, key management, key rotation, encrypted backup.

Based on this page: https://vercel.com/docs/concepts/observability/audit-log
it seems Vercel is meeting HIPAA for role-based management and audit logging.
but it may require a longer history time range?

Vercel may possibly already be meeting all standards for HIPAA, but just may have to reveal some features or reveal the ability to adjust parameters so the end-user can take control of their security and roles via the deployment settings within Vercels dashboard.

Keep in mind there is not really any organization that will fully certify an organization as fully HIPAA compliant, it is up to the organization to make that determination internally. To get a full checklist of what is needed to meet HIPAA security rules see pages 17 -> 53 of this document: https://nvlpubs.nist.gov/nistpubs/Legacy/SP/nistspecialpublication800-66r1.pdf

**Reply by jessethomson** [reply]
@leerob Even if I still have to use AWS for my database, S3, etc, having Vercel _hosting_ be HIPAA compliant with a standardized BAA would be a game-changer.

I'm currently in the process of setting up all the NextJS hosting infrastructure on AWS, and it's _painful_. Just offloading that piece alone would be amazing.

**Comment by raybotha** [comment]
This would be great, apps that need to render any PHI server-side or make any server-side API calls containing PHI can't be hosted on Vercel atm.

**Comment by mjyoung** [comment]
Also very interested in the ability to sign BAAs with Vercel. Any potential updates on this @swarnava? Currently self-hosting my Healthtech startup's Next.js app on GCP because of the need to have a signed BAA for anything related to PHI.

**Reply by swarnava** [reply]
Hi, sorry about the delayed reply. We are still exploring options. Currently we don't have an ETA!

**Comment by SawkaDev** [comment]
Interested in this also.

**Comment by davidtiede** [comment]
Vercel HIPAA compliance would be amazing, +1 to this request! 🔥

**Comment by CertainVictory** [comment]
+1

**Comment by jmarbutt** [comment]
This would be great!

**Comment by ebg1223** [comment]
Huge up vote for this! GCP and AWS all offer essentially free signed BAAs upon request. If vercel could extend this to their service, healthcare would move very quickly.

**Comment by irealva** [comment]
Very interested in Vercel being able to sign BAAs!

**Comment by michaeldeathless** [comment]
Also interested in this

**Comment by OSSRoger** [comment]
We at Neem are very interested in being able to include Vercel in the Vanta HIPAA process.

**Comment by brycedbjork** [comment]
Would love to sign a BAA with Vercel instead of GCP

**Comment by Wtrapp** [comment]
+1

**Comment by clairolzamcs** [comment]
+1

**Comment by raminsj13** [comment]
+1

**Comment by ebg1223** [comment]
@swarnava I noticed your update, and appreciate vercel looking into this! After working in health sector for some time, I wanted to share my own experience with AWS/GCP in case it's helpful. Just my experience, others can feel free to call me out or share their own.

I work on a team primarily developing applications which make specific roles / functions in healthcare much more streamlined, automated, and successful. In general, applications are extremely siloed, but often access to data is through standard protocols(FHIR), and the data is often stored/controlled by software from one of a handful of organizations.

Vercel would be an ideal place for these types of apps. They are usually commissioned by a small unit within a large healthcare system, or by a private, medium-sized organization. We can setup an "organization" for this project, giving all parties the necessary access and ensuring ownership is with the party who has rightful access to the data. 

Once they reach the phase where they are successfully making a difference for a group, they are able to offer the application to others, and because data is typically stored / accessed in the same way, it is very simple to bring many other users / systems into the application, and grow the user base dramatically.

What has made AWS/GCP so successful at getting these applications to be built cloud-first is the essentially free standard HIPAA compliance package. Anyone on GCP or AWS can request a signed BAA, which is standardized and limited to most core services. As others have said, it is likely the actual security provisions required for compliance are already in place for this type of agreement. We can spin up an organization for a project, develop, test, and deploy to a real world use case, and refine until it is ready for potentially expanding to larger organizations. 

These large organizations require more in depth compliance, and typically will sign large enterprise agreements with a cloud vendor for an app, which includes signing their own custom BAA, as well as other agreements, etc...

Tl;Dr- For vercel to attract health startups, as well as small to medium-sized health apps for enterprise, it would be amazing to have a limited, standardized BAA available on the vercel "Pro" tier, which would include core functionality, such as hosting, similar to AWS/GCP, either included with the tier or for a small fee. We are seeing several new app service providers taking this approach successfully for specific apis or components.

These apps tend to eventually grow to be deployed by large organizations, which would require enterprise level security as well as custom BAAs.

**Comment by chriscantu** [comment]
+1

**Comment by tylermzeller** [comment]
+1

**Comment by imgnx** [comment]
+1

**Comment by anthonyverducci** [comment]
This would be awesome. I'm currently working towards becoming HIPAA compliant, and keeping my project on Vercel would really simplify the process.

**Reply by Erik-Refdocs** [reply]
Same.

**Comment by JamieS1211** [comment]
+1

**Comment by anthonyverducci** [comment]
Hi all. I was gearing up to migrate my project to AWS hosting, and decided to reach out to Vercel's team to learn whether we'd have the option to sign a BAA in the coming months. I wanted to share their team's response in case it's helpful for others on this thread who are in a similar situation:

> Vercel supports healthcare customers by hosting their frontend without having to transmit or store PHI.
>
> Unless PHI is in the source code or in some part of dynamic asset registration, we will never have access to that information. There would be no need to act as a business associate or for Vercel to be HIPAA compliant.
>
> Please refer to our [privacy policy](https://link.vercel.com/ls/click?upn=fOHfzpdlPOqRaKdXEDcJp-2FyIykC5SwwCXT2hfigGX6liD1iFP52IalP3V5B-2B9E763UY2XrDUykiHMmLZ2J0XJCA6vzbRo-2Bzl-2FZV-2FkeVM19j8WNxffkPRKKBJOxFfS6JLFEhqUnTCDmHuClKxhTEcgVhizIcYS8OTwVdcIyWrpbNWXtIi8WyHIneYUd1kZxIYluRAWKiLY8laXDheYmZPp8S-2FyZBmZKSGgbkPK1eWaPopC0ClSJWMkfbsHUu4SG0PpQOcjPlDtm7s9QRyLNDiqh8um4kSxKznrjp9-2Bt3pkImDMSviNQmyIM1uuzjdsJGO3JMlShgqwurspFOKXPK9Yii9JRu7YpTmGp0q5G37uDPJ9uYdNJL05jbXCE4TzxvYCvUxldsotXOGQPEdtEMLWt88ZD2lwM75s0UBI9ctK6LB9b30hjG5Df8NNrSWfb-2BTS-2FX5HTeyiSiOXQAIHMiFFg-3D-3Djx6a_MTQeFU9OGQYuK17CNM-2FHMEvJcp0RtNJ15PPb1frmSdI-2BuYmQ6-2BJwrAnfi4lzWMJIl9SYQxzf-2B6aMjjKFgcvQkKDTN1Dumjs19v954agVfXGlTL1dWkrdbgGIqWZ4sB28Jr-2FPoh86LziKJsjKw490S58okFE9K7-2Be2zQqn8xByypQkBaEjKPcgpzBoOj1G0qZb-2BipfmYDHg4oVKdGpwnVAbO-2FojpfH9gPEu9Z3O4IJcHKGZ1e3tD4VuPAb8yPRyw3SOKGWs3-2Bdg0J76Gp0fUCpkaOUZ2A0cU3IdlYwQg8hxJFeVaWXdu0pDO1KcFb-2B7efk4Psu61K0L7potli47eRDFnrS80L9bNiUn6HfUPOTiRgOn2p9FDsSAbLniPvZAblZkl9kjnO-2F-2BJHpiCtYDAow07R8bCzujlHAtQpEnERwCXKcfpUNZg3OwapCm-2FJyel0wWSkA5gVz0TlCNPID5k6tNyNMgbJrHW-2Bnq6DB17QhEFVTqIsa6-2BAzjj0gJZNFmstdpxaYgG8-2BkUKu75IfEPXW9dHCAX8ejYuMpqcpxWfK4UyNAHSOKVrtN12cIK2xuPKt1SxJwaa2763jTtZLWQi-2BQJb9eImwYiryU-2BMyycWq1U-3D). If you have additional questions about HIPAA compliance or other aspects of our privacy policy, please reach out to [privacy@vercel.com](mailto:privacy@vercel.com).

I will consult our legal team to confirm, but as far as I understand, hosting supports healthcare applications with PHI, while Vercel's database products do not.

**Reply by raybotha** [reply]
> Hi all. I was gearing up to migrate my project to AWS hosting, and decided to reach out to Vercel's team to learn whether we'd have the option to sign a BAA in the coming months. I wanted to share their team's response in case it's helpful for others on this thread who are in a similar situation:
> 
> 
> 
> > Vercel supports healthcare customers by hosting their frontend without having to transmit or store PHI.
> 
> >
> 
> > Unless PHI is in the source code or in some part of dynamic asset registration, we will never have access to that information. There would be no need to act as a business associate or for Vercel to be HIPAA compliant.
> 
> >
> 
> > Please refer to our [privacy policy](https://link.vercel.com/ls/click?upn=fOHfzpdlPOqRaKdXEDcJp-2FyIykC5SwwCXT2hfigGX6liD1iFP52IalP3V5B-2B9E763UY2XrDUykiHMmLZ2J0XJCA6vzbRo-2Bzl-2FZV-2FkeVM19j8WNxffkPRKKBJOxFfS6JLFEhqUnTCDmHuClKxhTEcgVhizIcYS8OTwVdcIyWrpbNWXtIi8WyHIneYUd1kZxIYluRAWKiLY8laXDheYmZPp8S-2FyZBmZKSGgbkPK1eWaPopC0ClSJWMkfbsHUu4SG0PpQOcjPlDtm7s9QRyLNDiqh8um4kSxKznrjp9-2Bt3pkImDMSviNQmyIM1uuzjdsJGO3JMlShgqwurspFOKXPK9Yii9JRu7YpTmGp0q5G37uDPJ9uYdNJL05jbXCE4TzxvYCvUxldsotXOGQPEdtEMLWt88ZD2lwM75s0UBI9ctK6LB9b30hjG5Df8NNrSWfb-2BTS-2FX5HTeyiSiOXQAIHMiFFg-3D-3Djx6a_MTQeFU9OGQYuK17CNM-2FHMEvJcp0RtNJ15PPb1frmSdI-2BuYmQ6-2BJwrAnfi4lzWMJIl9SYQxzf-2B6aMjjKFgcvQkKDTN1Dumjs19v954agVfXGlTL1dWkrdbgGIqWZ4sB28Jr-2FPoh86LziKJsjKw490S58okFE9K7-2Be2zQqn8xByypQkBaEjKPcgpzBoOj1G0qZb-2BipfmYDHg4oVKdGpwnVAbO-2FojpfH9gPEu9Z3O4IJcHKGZ1e3tD4VuPAb8yPRyw3SOKGWs3-2Bdg0J76Gp0fUCpkaOUZ2A0cU3IdlYwQg8hxJFeVaWXdu0pDO1KcFb-2B7efk4Psu61K0L7potli47eRDFnrS80L9bNiUn6HfUPOTiRgOn2p9FDsSAbLniPvZAblZkl9kjnO-2F-2BJHpiCtYDAow07R8bCzujlHAtQpEnERwCXKcfpUNZg3OwapCm-2FJyel0wWSkA5gVz0TlCNPID5k6tNyNMgbJrHW-2Bnq6DB17QhEFVTqIsa6-2BAzjj0gJZNFmstdpxaYgG8-2BkUKu75IfEPXW9dHCAX8ejYuMpqcpxWfK4UyNAHSOKVrtN12cIK2xuPKt1SxJwaa2763jTtZLWQi-2BQJb9eImwYiryU-2BMyycWq1U-3D). If you have additional questions about HIPAA compliance or other aspects of our privacy policy, please reach out to [privacy@vercel.com](mailto:privacy@vercel.com).
> 
> 
> 
> I will consult our legal team to confirm, but as far as I understand, hosting supports healthcare applications with PHI, while Vercel's database products do not.

It's more of a technical issue with the way your app is coded than a legal one here, if PHI at any point touches their servers you need a BAA and HIPAA-compliance.

You would have to be careful with your architecture because Vercel blends frontend and backend processing in a way that's not always clear.

Edit: if your users are patients and not providers, it's much more unlikely Vercel would work because of the data they see regardless of your app.

**Reply by anthonyverducci** [reply]
Thanks for the explanation, @raybotha. Seems like I have some more learning to do here before making a decision.

In my case, the users are case managers and the app is set up to render all PHI client-side.

**Reply by raybotha** [reply]
> Thanks for the explanation, @raybotha. Seems like I have some more learning to do here before making a decision.
> 
> 
> 
> In my case, my users are case managers and all PHI is currently rendered client-side.

Good luck! Be sure to watch out for caching too.

**Reply by jessethomson** [reply]
> I will consult our legal team to confirm, but as far as I understand, hosting supports healthcare applications with PHI, while Vercel's database products do not.

I just want to clarify that this statement is only true in your specific circumstance, but not generally true.

As @raybotha said "if PHI at any point touches their servers you need a BAA and HIPAA-compliance."

So if you are able to use other HIPAA-compliant database and API products such that no PHI is ever sent to a Vercel server, then you could potentially get around this.

But for most apps that handle PHI, that PHI would absolutely be sent to Vercel's servers.

Unfortunately, this effectively means you can't host NextJS on Vercel in a HIPAA compliant way.

(I would love to be proven wrong as I am currently in the middle of the painful process of setting everything up on AWS for a HIPAA compliant app.)

**Reply by anthonyverducci** [reply]
Thanks, @jessethomson!

My current solution is to use Supabase's HIPAA-compliant database and APIs through their "Teams" plan. It's expensive, but it'll make my life much easier since my project already uses Supabase and I prefer its developer experience over Amazon's database products.

In my Next.js / Vercel implementation, I only fetch PHI on the client side. So this setup should work in theory. However, I would definitely feel more confident if I could sign a BAA with Vercel, which is  why I'm still actively considering AWS.

It's also worth mentioning for others in this thread that Vercel updated their [HIPAA page on their website](https://vercel.com/guides/is-vercel-hipaa-compliant) — they directly address the comments above regarding BAAs.

Good luck with your project @jessethomson!

**Reply by raybotha** [reply]
I don't think the Vercel page addresses it well. It says "Vercel will not have access to or see any customer healthcare data since we terminate SSL/TLS" which I'm not sure what it's trying to say. Usually terminating SSL would imply the opposite, that you ARE processing unencrypted data at that point. Which Vercel would be doing if you run any server-side code that involves health data.

The way most people use hosting for Next.js is not as a "dumb" S3 bucket of assets (which would not require HIPAA compliance) but as a server which can make server-side calls to databases, APIs, caches, etc all of which would make Vercel a data processor under HIPAA.

**Reply by jessethomson** [reply]
> The way most people use hosting for Next.js is not as a "dumb" S3 bucket of assets, ... but as a server ... which would make Vercel a data processor under HIPAA

Exactly. Unfortunately, it seems there is no way to use any server functionality with NextJS on Vercel and be HIPAA compliant.

> I don't think the Vercel page addresses it well.

Agreed. At best, it's confusing. At worst, it's misleading.

**Comment by courtstarr** [comment]
Adding a request for HIPAA Compliance as well. I feel Vercel is likely close to being able to support this with certain configurations (e.g. AWS RDS or Supabase for db). It would be a huge win for many companies I work with to have Next.js + Vercel to be an option for development & hosting.

**Comment by justinkibu** [comment]
+1

**Comment by mjyoung** [comment]
Cmon Vercel please make this a thing in 2024 :)

**Comment by conoremclaughlin** [comment]
+1 - it's very tempting to host our Frontend on Vercel given it's client-side only, but I'm not comfortable with us potentially letting some server-side calls slip in with PHI. Better for the worst case to still be compliant by hosting on Azure even if the DX is truly terrible.

**Comment by swarnava** [comment]
**We are now PCI compliant for ecommerce teams: https://vercel.com/blog/pci-compliance-for-ecommerce-teams**

We haven't forgot about HIPAA Compliance and our team is working diligently to support this. More details soon, hopefully! 😄

**Reply by jessethomson** [reply]
Oh man, of course you give me hope for Vercel HIPAA right after I decided to migrate from AWS to fly.io! 😅 

But this is an exciting development! I would love to have Vercel as an option and am willing to forgo my thousands of dollars in AWS credits for the better DX 😆 

Decisions, decisions...

**Comment by Robertorc83** [comment]
+1

**Comment by 0arm** [comment]
+1000

**Comment by shivasuri** [comment]
Any update on this? We currently use Vercel Postgres but we may have to switch to a posgres instance from a company that will support signing a BAA

**Comment by leerob** [comment, marked as answer]
Good news! Vercel now [supports HIPAA compliance](https://vercel.com/blog/vercel-supports-hipaa-compliance) and can sign Business Associate Agreements (BAAs) with enterprise customers.

Please [reach out to our team](https://vercel.com/contact/sales) if you'd to talk about this.

**Reply by courtstarr** [reply]
BOOM! Thanks everyone who helped make this a reality

**Comment by 0arm** [comment]
So you have to spend at least 4 figures per month with Vercel as an enterprise customer for a BAA?

**Reply by leerob** [reply]
That is correct, this is for Enterprise customers only.
